dfx/index.ts

```diff
@@ -60,8 +60,38 @@ export function getIdentityName(): string {
     return execSync(`dfx identity whoami`).toString().trim();
 }
 
-export function generateIdentity(name: string) {
-    execSync(`dfx identity new ${name} --storage-mode plaintext`);
+type StorageMode = 'keyring' | 'password-protected' | 'plaintext';
+
+export function generateIdentity(
+    name: string,
+    storageModeCool?: StorageMode
+): Buffer {
+    console.info();
+    console.info(`Generating identity ${name}`);
+    const storageMode = determineStorageMode(storageModeCool);
+    if (storageMode === undefined) {
+        console.info(`You may be prompted to create a password for ${name}`);
+        console.info();
+        return execSync(`dfx identity new ${name}`, {
+            stdio: ['inherit', 'pipe', 'inherit']
+        });
+    }
+    if (storageMode === 'password-protected') {
+        console.info(`You will be prompted to create a password for ${name}`);
+        console.info();
+    }
+    return execSync(`dfx identity new ${name} --storage-mode ${storageMode}`, {
```

Comment:
Seems like the user won't get the warning here about being prompted for their password

Reply:
If we get to this point they won't be prompted for their password since it will be either keychain or plaintext, neither of those use a password